File: Project-Euler/Solutions-Simon/euler7.jl
#=
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?
=#
#Answer = 104_743

function isPrime(x)
  divisor = 2
  prime = true
  while divisor < x
    if x % divisor == 0
      #@printf "%d mod %d == 0\n" x divisor
      prime = false
      break
    end
    divisor += 1
  end
  return prime
end
#println(isPrime(7))

function findPrime()
  arrayPrimes = Int64[]
  counter = 0
  number = 2
  primeNumber = 10001
  while counter < primeNumber
    if isPrime(number)
      counter += 1
      #println(number)
      if counter == primeNumber
        break
      end
    end
    number += 1
  end
  push!(arrayPrimes, number)
  #println(size(arrayPrimes))
  println(arrayPrimes[end])
end

findPrime()

File: Benchmarking/stats/euler7/euler7.java
import java.util.*;

public class euler7
{
	public static void main(String[] args)
	{
		int primeNumber = Integer.parseInt(args[0]);
		System.out.println(findPrime(primeNumber));
	}
	
	private static int findPrime(int primeNumber)
	{
		List<Integer> arrayPrimes = new ArrayList<>();
		int counter = 0;
		int number = 2;
		while(counter < primeNumber)
		{
			if(isPrime(number))
			{
				counter += 1;
				if(counter == primeNumber)
				{
					break;
				}
			}
			number += 1;
		}
		if(counter == primeNumber)
		{
			return number;
		}
		return 0;
	}
	
	private static boolean isPrime(int x)
	{
		int divisor = 2;
		boolean prime = true;
		while(divisor < x)
		{
			if(x % divisor ==0)
			{
				prime = false;
				break;
			}
			divisor += 1;
		}
		return prime;
	}
}
